### DISCUSSION TITLE
Debugging outside cluster using VS Code

### DISCUSSION THREAD

**Discussion by v-shenoy** [kedacore/keda#2887] [Q&A / Need Help]
It would be helpful if the steps to debug using VS Code were documented alongside the build instructions in `BUILD.md`.  I have been able to debug the KEDA Operator  with the following `launch.json` and running the `main.go` file.

```json
{
    "configurations": [
        {
            "name": "Launch file",
            "type": "go",
            "request": "launch",
            "mode": "debug",
            "program": "${file}",
            "env": {"WATCH_NAMESPACE": ""}
        }
    ]
}
```

But haven't been able to figure out how to do so with the metrics-apiserver.

@kedacore/keda-core-contributors 

Edit - Accidentally created this in the docs repo, and am unable to transfer it to the main one.

**Comment by zroubalik** [comment]
Could you please add the example `launch.json` to do BUILD.md?

I haven't tried to run debugger on Metrics Server yet, so not sure about the configuration. The point is, that it needs to be targeted by `APIService` resource in k8s.
You can check this, but I concur it doesn't give you all the info you need: https://github.com/kubernetes-sigs/custom-metrics-apiserver#clone-and-build-the-testing-adapter

But if you are able to setup the debugging, would be great if you can share this.

**Reply by v-shenoy** [reply]
Sure, I will add the instructions for the `operator` in `BUILD.md`.

As for the metrics server, I will try and go through the link. But I am not an expert in Kubernetes, so I am not sure if I will get it. Thanks for the info, however.
